packages/replay-next/components/console/hooks/usePointInstances.ts

```diff
@@ -0,0 +1,162 @@
+      };
+    },
+    () =>
+      getMemoizationToken({
+        client,
+        focusRange,
+        pointBehaviors,
+        points,
+      }),
+    () =>
+      getMemoizationToken({
+        client,
+        focusRange,
+        pointBehaviors,
+        points,
+      })
+  );
+
+  // Lazily fetch any hit points that weren't already fetched yet
+  // useSyncExternalStore will ensure this component re-renders once values have loaded
+  useEffect(() => {
+    if (focusRange !== null) {
+      points.forEach(point => {
+        hitPointsForLocationCache.readAsync(
+          client,
+          toPointRange(focusRange),
+          point.location,
+          point.condition
+        );
+      });
+    }
+  });
+
+  return useMemo<PointInstance[]>(() => {
+    const pointInstances: PointInstance[] = [];
+
+    // Reference this so the rules-of-hooks won't complain.
+    memoizationToken;
+
+    if (focusRange == null) {
+      return EMPTY_ARRAY;
+    } else {
+      points.forEach(point => {
+        const pointBehavior = pointBehaviors[point.key];
+        if (pointBehavior?.shouldLog === POINT_BEHAVIOR_ENABLED) {
+          // Don't suspend here; use a cached value if there is one
+          const value = hitPointsForLocationCache.getValueIfCached(
+            client,
+            toPointRange(focusRange),
+            point.location,
+            point.condition
+          );
+          if (value) {
+            const [hitPoints, status] = value;
+
+            switch (status) {
+              case "too-many-points-to-find":
+              case "too-many-points-to-run-analysis": {
+                // Don't try to render log points if there are too many hits.
+                break;
+              }
+              default: {
+                hitPoints.forEach(hitPoint => {
+                  pointInstances.push({
+                    point,
+                    timeStampedHitPoint: hitPoint,
+                    type: "PointInstance",
+                  });
+                });
+                break;
+              }
+            }
+          }
+        }
+      });
+    }
+
+    return pointInstances;
+  }, [client, focusRange, memoizationToken, pointBehaviors, points]);
+}
+
+function getMemoizationToken({
+  client,
+  focusRange,
+  pointBehaviors,
+  points,
+}: {
+  client: ReplayClientInterface;
+  focusRange: TimeStampedPointRange | null;
+  pointBehaviors: PointBehaviorsObject;
+  points: Point[];
+}) {
+  const keys: string[] = [];
+
+  if (focusRange !== null) {
+    points.forEach(point => {
+      const pointBehavior = pointBehaviors[point.key];
+      if (pointBehavior?.shouldLog === POINT_BEHAVIOR_ENABLED) {
+        const cachedValue = hitPointsForLocationCache.getValueIfCached(
+          client,
+          toPointRange(focusRange),
+          point.location,
+          point.condition
+        );
+        if (cachedValue != null) {
+          keys.push(point.key);
```

Comment:
Delta test (`should-support-conditional-log-points.ts`) caught a small regression in this also. Will address shortly.